arlas-core/src/main/java/io/arlas/server/utils/ElasticClient.java

```diff
@@ -0,0 +1,271 @@
+            request.fetchSourceContext(fetchSourceContext);
+            GetResponse hit = client.get(request, RequestOptions.DEFAULT);
+            return hit.getSourceAsString();
+        } catch (IOException e) {
+            processException(e, index);
+            return null;
+        }
+    }
+
+    public SearchResponse search(SearchRequest request) throws ArlasException {
+        try {
+            return client.search(request, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            processException(e, request.indices().toString());
+            return null;
+        }
+    }
+
+    public SearchResponse searchScroll(SearchScrollRequest request) throws ArlasException {
+        try {
+            return client.scroll(request, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            processException(e, request.getDescription());
+            return null;
+        }
+    }
+
+    public ClearScrollResponse clearScroll(ClearScrollRequest request) throws ArlasException {
+        try {
+            return client.clearScroll(request, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            processException(e, request.getDescription());
+            return null;
+        }
+    }
+
+    public DeleteResponse delete(String index, String ref) throws ArlasException {
+        try {
+            DeleteRequest request = new DeleteRequest(index, ref);
+            return client.delete(request, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            processException(e, index);
+            return null;
+        }
+    }
+
+    public AcknowledgedResponse deleteIndex(String index) throws ArlasException {
+        try {
+            return client.indices().delete(new DeleteIndexRequest(index), RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            processException(e, index);
+            return null;
+        }
+    }
+
+    public IndexResponse index(String index, String id, String source) throws ArlasException {
+        try {
+            IndexRequest request = new IndexRequest(index).id(id);
+            request.source(source, XContentType.JSON);
+            return client.index(request, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            processException(e, index);
+            return null;
+        }
+    }
+
+    private void processException(IOException e, String index) throws ArlasException {
+        if (e instanceof ResponseException) {
+            if (((ResponseException) e).getResponse().getStatusLine().getStatusCode() == 404) {
+                throw new NotFoundException("Index " + index + " does not exist.");
+            }
+        }
+        LOGGER.warn("Exception while communicating with ES: " + e.getMessage(), e);
+        throw new InternalServerErrorException(e.getMessage());
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
+        super.finalize();
```

Comment:
`Finalize` is not always called, will it be OK for the resources?

Reply:
Considering that the REST client is not permanently connected to ES (http requests), it will have no impact on ES. And since the finalize method is supposed to be called when stopping the application, I assume it will not be an issue.